cmake/ProtoBuf.cmake

```diff
@@ -94,7 +108,7 @@ endif()
 # convention we will use SYSTEM inclusion path.
 get_target_property(__tmp protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
 message(STATUS "Caffe2 protobuf include directory: " ${__tmp})
-include_directories(SYSTEM ${__tmp})
+include_directories(BEFORE SYSTEM ${__tmp})
```

Comment:
For the error message that you are encountering on cuda, I suspect this is the cause, but I don't know the underlying reason. Maybe this is related:

https://github.com/facebookresearch/Detectron/blob/master/lib/CMakeLists.txt#L37-L39

Reply:
It think it was caused by my using

`target_link_libraries(caffe2 PUBLIC protobuf::libprotobuf)` 

instead of 

`target_link_libraries(caffe2 INTERFACE protobuf::libprotobuf)`

like you were, so I've switched it back to see if all the builds and tests pass. Cross fingers.